src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1791,9 +1771,10 @@ private IEnumerable<TypeInferenceInfo> InferTypeInYieldStatement(YieldStatementS
                     return SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>();
                 }
 
-                var memberSymbol = GetDeclaredMemberSymbolFromOriginalSemanticModel(SemanticModel, yieldStatement.GetAncestorOrThis<MemberDeclarationSyntax>());
+                var declaration = yieldStatement.FirstAncestorOrSelf<SyntaxNode>(n => n.IsReturnableConstruct());
+                var memberSymbol = GetDeclaredMemberSymbolFromOriginalSemanticModel(declaration);
 
-                var memberType = GetMemberType(memberSymbol);
+                var memberType = GetMemberType(memberSymbol, out _);
```

Comment:
💭 The new out parameter (`isAsync`) is currently not needed in this place, but will be when we get async streams. Should I maybe file an issue so that we don't forget to update the logic here? cc @jcouv

Reply:
Thanks @Neme12 
I can add a note to the async-streams [work items](https://github.com/dotnet/roslyn/issues/24037). Is there a fixer or scenario that is most relevant to cover?

Reply:
@jcouv It would be for generating methods or variables from yield returning in an async iterator. Instead of unwrapping the IEnumerable, it should unwrap IAsyncEnumerable.

I can take of fixing that if you ping me when both this PR and async iterators are merged.